src/System.Management.Automation/engine/parser/ast.cs

```diff
@@ -7863,6 +7869,11 @@ public MemberExpressionAst(IScriptExtent extent, ExpressionAst expression, Comma
             this.Member = member;
             SetParent(member);
             this.Static = @static;
+
+            if (genericTypes?.Any() == true)
+            {
+                this.GenericTypeArguments = new ReadOnlyCollection<ITypeName>(genericTypes.ToArray());
```

Comment:
Can we avoid these LINQ?
Maybe always initialize GenericTypeArguments with empty collection if right argument is null? This allows us remove many null checks.